tests/integration-tests/tests/ad_integration/test_ad_integration.py

```diff
@@ -120,6 +120,18 @@ def _add_file_to_zip(zip_file, path, arcname):
         zip_file.writestr(zinfo, input_file.read())
 
 
+def add_tag_to_stack(stack_name, key, value):
+    cfn = boto3.resource("cloudformation")
+    stack = cfn.Stack(stack_name)
+    stack.update(
```

Comment:
This implies a stack update operation that takes time. Better to avoid it just to add a tag if not strictly necessary. Why do we need to update a stack with tags. The mechanism should be applied only on brand new stack not on existing one, I guess.

Reply:
The tag is added to the new AD stacks when they are created, but the vpc stack has already been created without the tag, so that is why if we create new AD stacks, we need to update the vpc stack tags.